src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
-            final String serviceEventType, final String serviceId,
-            final String serviceName, final String serviceVersion,
-            final String serviceData) {
+
+    public static CloudEvent createRepositoryDeletedEvent(
+            final String eventType, final String id,
+            final URI source, final String name,
+            final String owner, final URI url,
+            final URI viewUrl, final String data) {
         CloudEvent ceToSend =
-                buildCloudEvent(serviceEventType, serviceData)
-                .withExtension("serviceid", serviceId)
-                .withExtension("servicename", serviceName)
-                .withExtension("serviceversion", serviceVersion)
-                .build();
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("name", name)
+                        .withExtension("owner", owner)
+                        .withExtension("url", url)
+                        .withExtension("viewurl", viewUrl)
+                        .build();
         return ceToSend;
     }
 
+    /**
+     * Creates a continuous delivery branch created event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * branch created event extensions
+     */
+    public static CloudEvent createBranchCreatedEvent(
+            final String eventType, final String id,
+            final URI source, final Repository repository,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery branch deleted event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * branch deleted event extensions
+     */
+
+    public static CloudEvent createBranchDeletedEvent(
+            final String eventType, final String id,
+            final URI source, final Repository repository,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())

```

Comment:
please use `jackson-databind` to write this `repository` as String, say objectMapper.writeValueAsString(repository)